packages/data-stores/src/add-ons/hooks/use-add-on-prices.ts

```diff
@@ -17,7 +18,10 @@ export const createAddOnPriceKey = ( { productSlug, quantity }: AddOnMeta ) => {
 };
 
 export const useAddOnPrices = ( addOnMetas: AddOnMeta[] ) => {
-	const productSlugs = addOnMetas.map( ( { productSlug } ) => productSlug );
+	const productSlugs = useMemo(
+		() => addOnMetas.map( ( { productSlug } ) => productSlug ),
+		[ addOnMetas ]
+	);
```

Comment:
`.map` creates a new array on every call. So the `useMemo` function below, which uses `productSlugs` as a dependency, could not memoize the return value since the reference changed on every render.